lib/core/engine_clang.cpp

```diff
@@ -339,45 +332,58 @@ namespace
   {
     std::vector<std::string>
       args{
-        "-Wfatal-errors",
         "-iquote", ".",
         "-x", "c++-header",
         "-I", internal_dir_
       };
 
-    if (env_detector_.on_windows())
+    if (UseInternalTemplight)
     {
-      args.push_back("-fno-ms-compatibility");
-      args.push_back("-U_MSC_VER");
-    }
+      args.push_back("-Wfatal-errors");
 
-    if (!cpp_standard_set(config_.extra_clang_args))
-    {
-      args.push_back("-std=c++0x");
-    }
+      if (env_detector_.on_windows())
+      {
+        args.push_back("-fno-ms-compatibility");
+        args.push_back("-U_MSC_VER");
+      }
+
+      if (!cpp_standard_set(config_.extra_clang_args))
+      {
+        args.push_back("-std=c++0x");
+      }
+
+      if (!max_template_depth_set(config_.extra_clang_args))
+      {
+        args.push_back(set_max_template_depth(256));
+      }
   
-    if (!max_template_depth_set(config_.extra_clang_args))
-    {
-      args.push_back(set_max_template_depth(256));
-    }
+      const std::vector<std::string> include_path =
+        determine_include_path(clang_path_, env_detector_, logger_);
+      args.reserve(args.size() + include_path.size());
+      for (const std::string& p : include_path)
+      {
+        args.push_back("-I" + p);
+      }
 
-    const std::vector<std::string> include_path =
-      determine_include_path(clang_path_, env_detector_, logger_);
-    args.reserve(args.size() + include_path.size());
-    for (const std::string& p : include_path)
+      args.insert(
+        args.end(),
+        config_.extra_clang_args.begin(),
+        config_.extra_clang_args.end()
+      );
+    }
+    else if (config_.extra_clang_args.size() > 1)
     {
-      args.push_back("-I" + p);
+      args.insert(
+        args.end(),
+        config_.extra_clang_args.begin() + 1,
+        config_.extra_clang_args.end()
+      );
     }
 
-    args.insert(
-      args.end(),
-      config_.extra_clang_args.begin(),
-      config_.extra_clang_args.end()
-    );
-
     return args;
   }
 
+  template <bool UseInternalTemplight>
```

Comment:
The gcc in my dev env (4.8.4) doesn't seem to be able to handle parameter packs in lambdas. I haven't tried Visual C++, but I'd leave it the way it is for now to avoid having to upgrade compilers on all platforms.

Note that I don't expect to have similar issues with other engines, as Clang is special case because of it being the "internal" compiler.